technology-choices.md

```diff
@@ -50,6 +50,7 @@ Helpers & tooling
 - docker (and Dockerfiles) for creating runnable, publishable images
   - make the image so that `docker run XXX` (with environment variables)
     brings up a working service
+    - Speficially for transpiled and packaged ES/React UIs, make sure that the environment variables are respected at runtime and not at compile time, and that the transpiled application is pre-built into the docker image (doesn't need to be rebuilt in the production environment). This means that the environment variables needed in the browser app should be read by the application's node server and transmitted to the browser runtime environment by embedding them into the server generated HTML or by providing an API for them. Make sure to whitelist and transmit only those environment variables which do not contain security-critical secrets or other sensitive data.
```

Comment:
Right now most / all of the projects started in the autumn of 2019 and later are based on create-react-app and not ejected - and they are hosted by nginx without any server side code. kukkuu-ui is one example. 

I think our setup is not in line with this requirement? I hope we can change this so that it fits with the unejected create-react-apps since we're avoiding a lot of complexity this way. 

We've been able to do this without putting any sensitive data in the repo or into the environments.

Reply:
very good point.

Reply:
Does create-react-app always serve static pages?

Reply:
I think you can use whichever web server with create-react-app applications, so you can interpolate configuration values into the HTML as documented here: https://create-react-app.dev/docs/title-and-meta-tags#injecting-data-from-the-server-into-the-page

Reply:
Let's discuss this in detail somewhere. I'd just like to point out that elsewhere we've been conserned about SEO and search engine / social media visibility of UIs, and having them purely static without server side rendering also contradicts that requirements unless we use a separate SaaS service for prerendering these client-side-javascript-only apps.